lib/src/main/kotlin/com/swmansion/starknet/provider/gateway/GatewayProvider.kt

```diff
@@ -122,10 +122,9 @@ class GatewayProvider(
 
     private fun transformResponseForTransactionAndSerialize(body: String): Transaction {
         try {
-            val transformedResponse = json.parseToJsonElement(body).jsonObject["transaction"]
-            val encodedResponse = json.encodeToString(transformedResponse)
-            return json.decodeFromString(TransactionPolymorphicSerializer, encodedResponse)
-        } catch (e: IllegalArgumentException) {
+            val transformedResponse = json.parseToJsonElement(body).jsonObject.getValue("transaction")
+            return json.decodeFromJsonElement(TransactionPolymorphicSerializer, transformedResponse)
+        } catch (e: Exception) {
```

Comment:
Isn't it too broad?

Reply:
I though the same too. However, kotlin doesn't (why???) allow catching multiple exceptions in one catch. And `getValue` raises `NoSuchElementException`. 

We could have two catch blocks (duplications), or it can be changed to `json.parseToJsonElement(body).jsonObject["transaction"] :? throw IllegalArgumentException()` so we have a single catch.

Reply:
Hmm I see. Maybe we could catch Exception and then decide what to do in `when` block? It is not perfect, but we could still throw `RequestFailedException` with specific, custom info, so maybe it's worth it